src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -199,22 +216,26 @@ def extract_strings(self) -> Mapping[int, Set[str]]:
                 self.skipped_offsets.add(partition_offset)
                 continue
 
-            parsed_strings = {
+            strings_in_message = {
                 metric_name,
                 *tags.keys(),
             }
 
             if self.__should_index_tag_values:
-                parsed_strings.update(tags.values())
+                strings_in_message.update(tags.values())
 
-            org_strings[org_id].update(parsed_strings)
+            strings[use_case_id][org_id].update(strings_in_message)
 
-        string_count = 0
-        for org_set in org_strings:
-            string_count += len(org_strings[org_set])
-        metrics.gauge("process_messages.lookups_per_batch", value=string_count)
+        metrics.gauge(
+            "process_messages.lookups_per_batch",
+            value=sum(
+                len(parsed_strings)
+                for use_case in strings.values()
+                for parsed_strings in use_case.values()
+            ),
+        )
 
-        return org_strings
+        return strings
 
     @metrics.wraps("process_messages.reconstruct_messages")
     def reconstruct_messages(
```

Comment:
Since we've now added the `use_case_id` field to the message payload upon instantiation of a Batch (see `_extract_messages`), we're going to need to delete the lines in `reconstruct_messages` that add the `use_case_id` onto the payload in a hardcoded manner (this was the old way of adding `use_case_id` to the message).